### PR TITLE
factor twobit store out into npm module

### PR DESCRIPTION
**Pull request by rbuels**
out2.2bit now loads in about 9 seconds on my machine. Still not awesome, but better than it was.  And the code is quite a bit cleaner now.

### CODE REVIEW

**Comment by rbuels** [comment]
This should fix #1116

**Comment by cmdcolin** [comment]
I'm trying out on a wheat 2bit file from Nathan Haigh and it is giving some problems with alternating blocks being loaded? I also tried it on volvox.2bit and it seemed to work fine so I'd have to do a little more debugging to find out why it is weird on this specific data. If you have that wheat genome maybe try converting it to 2bit to test


screenshot-localhost-2018 07 28-14-11-51

**Comment by rbuels** [comment]
you talking about @nathanhaigh's 161010_Chinese_Spring_v1.0_pseudomolecules.fasta?

**Comment by cmdcolin** [comment]
yep :)

**Comment by nathanhaigh** [comment]
@cmdcolin @rbuels According to the format description:

"A .2bit file stores multiple DNA sequences (up to 4 Gb total) in a compact randomly-accessible format."

So I guess it won't support references that size?

**Comment by nathanhaigh** [comment]
Though I also see that `faToTwoBit` has a `-long` option to overcome this limitation.

**Comment by rbuels** [comment]
ah, a weird bug.  fixed it in @gmod/twobit 1.0.2.  https://github.com/GMOD/twobit-js/commit/69ecf95ba6f95bd5f2aa691974aeb6a79a5c206f was the fix

**Comment by rbuels** [comment]
@nathanhaigh your wheat sequence fit OK in a .2bit file, it only ended up being 3.5G. starting to max out the format a bit though.  they need to publish an upgraded version of the format with bigger fields for the file offsets, at minimum.

**Comment by rbuels** [comment]
oh that's what the `-long` option does, 64-bit offsets.  I need to upgrade
the twobit-js to support that then.

**Comment by nathanhaigh** [comment]
Oh...I thought the limit would likely be Gbp (number of nucleotides) not GB as in file size.

**Comment by rbuels** [comment]
Alright I think it should be good now, could you try it again and confirm?

File: src/JBrowse/Store/SeqFeature/TwoBit.js [hunk 1]
```diff
@@ -33,60 +27,15 @@ return declare([ SeqFeatureStore, DeferredFeaturesMixin], {
      */
     constructor: function( args ) {
 
-        var blob = args.blob || new XHRBlob( this.resolveUrl( args.urlTemplate || 'data.2bit' ), { expectRanges: true } );
```
Comment (cmdcolin) [inline src/JBrowse/Store/SeqFeature/TwoBit.js]: any reason why expectRanges removed?

Comment (rbuels) [reply src/JBrowse/Store/SeqFeature/TwoBit.js]: Nope, fixed.

**Comment by rbuels** [comment]
I haven't been able to reproduce the "start cannot be negative" problem. Do you have a recipe for getting that error?

**Comment by cmdcolin** [comment]
It looks like a int overflow somewhere from signed int 2^31...

Wondering if maybe there is a unsigned int in the twobit-js code

e.g.

```
diff --git a/src/twoBitFile.js b/src/twoBitFile.js
index 7f72555..9aace85 100644
--- a/src/twoBitFile.js
+++ b/src/twoBitFile.js
@@ -120,7 +120,7 @@ class TwoBitFile {
         }),
       record1: new Parser()
         .endianess(endianess)
-        .int32('dnaSize')
+        .uint32('dnaSize')
         .int32('nBlockCount'),
       record2: new Parser()
         .endianess(endianess)
```

**Comment by cmdcolin** [comment]
Here's one random tidbit possible related to the above (uint vs int)

If I do `twoBitInfo wheat.2bit output.tab` I get

```
chr1A   594102056
chr1B   689851870
chr1D   495453186
chr2A   780798557
chr2B   801256715
chr2D   651852609
chr3A   750843639
chr3B   830829764
chr3D   615552423
chr4A   744588157
chr4B   673617499
chr4D   509857067
chr5A   709773743
chr5B   713149757
chr5D   566080677
chr6A   618079260
chr6B   720988478
chr6D   473592718
chr7A   736706236
chr7B   750620385
chr7D   638686055
chrUn   480980714
```

If I make a small nodejs script

```
require('regenerator-runtime/runtime');

const { TwoBitFile } = require('@gmod/twobit')
const t = new TwoBitFile({
      path: require.resolve('./seq/out.2bit'),
})
t.getSequenceSizes().then((response) => {
    Object.keys(response).forEach(key => {
        console.log(key+"\t"+response[key]);
    });
})

```
Then I get

```
chr1A   594102056
chr1B   689851870
chr1D   495453186
chr2A   780798557
chr2B   801256715
chr2D   651852609
chr3A   750843639
chr3B   830829764
chr3D   615552423
chr4A   744588157
chr4B   673617499
chr4D   509857067
chr5A   709773743
chr5B   440477507
chr5D   22
chr6A   1919443717
chr6B   1661272064
chr6D   148846134
chr7A   627459121
chr7B   1093825128
chr7D   1661281852
chrUn   1919443717

```

**Comment by cmdcolin** [comment]
Looking good!

**Pull request merged**